rustls/src/quic.rs

```diff
@@ -745,6 +762,24 @@ pub trait PacketKey: Send + Sync {
         payload: &'a mut [u8],
     ) -> Result<&'a [u8], Error>;
 
+    /// Decrypt a multipath QUIC packet
+    ///
+    /// Takes a `path_id` and `packet_number`, used to derive the nonce; the packet `header`, which is used as
+    /// the additional authenticated data; and the `payload`. The authentication tag is returned if
+    /// encryption succeeds.
+    ///
+    /// If the return value is `Ok`, the decrypted payload can be found in `payload`, up to the
+    /// length found in the return value.
```

Comment:
Does this need to be reworded? I'm getting thrown by the "the length found in the return value part" since it looks like the signature returns `&[u8]` in the success case and the impl is giving `Ok(&payload[..plain_len])` where from the docs alone I'd assume it'd return `plain_len` and the caller has to dig it out of `payload`.

Reply:
this comes from the existing method https://github.com/rustls/rustls/blob/e2f6115548b54e15ec52517208e4b4c6f27aa7db/rustls/src/crypto/ring/quic.rs#L143-L150

Reply:
ok after checking it I agree it's misleading. What about
```rust
/// If the return value is `Ok`, it references the slice of `payload` containing the decrypted payload
```

suggestions welcome

Reply:
How about something like:

> /// On success, returns the slice of `payload` containing the decrypted data